Yosemite/YosemiteTests/Stores/Order/OrdersUpsertUseCaseTests.swift

```diff
@@ -327,6 +327,54 @@ final class OrdersUpsertUseCaseTests: XCTestCase {
         XCTAssertEqual(storageOrder.appliedGiftCards?.map { $0.toReadOnly() }.sorted(by: { $0.amount > $1.amount }),
                        [giftCard1, giftCard2])
     }
+
+    func test_it_persists_order_attribution_info_in_storage() throws {
+        // Given
+        let attributionInfo = OrderAttributionInfo.fake()
+        let order = makeOrder().copy(siteID: 3, orderID: 11, attributionInfo: attributionInfo)
+        let useCase = OrdersUpsertUseCase(storage: viewStorage)
+
+        // When
+        useCase.upsert([order])
+
+        // Then
+        let storageAttributionInfo = try XCTUnwrap(viewStorage.loadOrder(siteID: 3, orderID: 11)?.attributionInfo)
+        XCTAssertEqual(storageAttributionInfo.toReadOnly(), attributionInfo)
+    }
+
+    func test_it_replaces_existing_order_attribution_info_in_storage() throws {
+        // Given
+        let originalAttributionInfo = OrderAttributionInfo.fake().copy(source: "admin")
+        let order = makeOrder().copy(siteID: 3, orderID: 11, attributionInfo: originalAttributionInfo)
+        let useCase = OrdersUpsertUseCase(storage: viewStorage)
+        useCase.upsert([order])
+
```

Comment:
Would it make sense to add a confirmation check to the test, so we're sure that the initial data is upserted correctly? 
```suggestion
        let originalStorageAttributionInfo = try XCTUnwrap(viewStorage.loadOrder(siteID: 3, orderID: 11)?.attributionInfo)
        XCTAssertEqual(originalStorageAttributionInfo.toReadOnly(), originalAttributionInfo, "Precondition: Original attribution info has been upserted to storage.")
```

Reply:
I preferred not to add a confirmation check as we have a dedicated unit test `test_it_persists_order_attribution_info_in_storage` to test this behaviour. IMO, testing for one focused behaviour in a single unit test would help us immediately understand what failed. What do you think?

Reply:
Sounds good to me!